code/client/launcher/Bootstrap.cpp

```diff
@@ -73,6 +74,19 @@ bool VerifyViability();
 
 extern void ResetUpdateChannel();
 
+void MessageBox_OnFailedRequestURL(int result)
+{
+	switch(result)
+	{
+	case 7: // CURL ERROR 7 Failed to connect, request is blocked by a firewall or similar.
+		MessageBox(NULL, va(L"An error (%i, %s) occurred while checking the bootstrapper version. " CONTENT_URL_WIDE L" is being blocked by your firewall (%s), to fix this you need to allow " PRODUCT_NAME L" through your firewall or disable it.", result, ToWide(DL_RequestURLError()), os::GetActiveFirewallName()), L"O\x448\x438\x431\x43A\x430", MB_OK | MB_ICONSTOP);

```

Comment:
If firewall is 'Unknown' (or if Windows [Defender] Firewall shows up as a result here) maybe not showing such in the message could be better - 'your firewall' seems better than 'your firewall (Unknown)'